### PR TITLE
[Model Monitor] - Add model monitoring component CI which runs unit test.

### PR DESCRIPTION
**Pull request by mastloui-msft**
- Adding a new workflow file which runs the model monitoring unit tests.
- Workflow runs only for changes under the model_monitoring/ folder and changes to the file itself.

### CODE REVIEW

File: .github/workflows/model-monitoring-ci.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  id-token: write
+
+defaults:
+  run:
+    shell: cmd
+
+jobs:
+  run-e2e-tests:
+    name: Run Monitoring Component Tests
+    runs-on: windows-latest
```
Comment (elesel) [inline .github/workflows/model-monitoring-ci.yml:35]: Does it specifically need Windows? This will be the first non-Linux test in the repo, I believe.

**Review by elesel — DISMISSED** [review]

File: .github/workflows/model-monitoring-ci.yml (hunk 1 above)
Comment (mastloui-msft) [reply .github/workflows/model-monitoring-ci.yml:35]: let me see if ubuntu works! It used to need windows due to the azure cli task in ADO :)

Comment (mastloui-msft) [reply .github/workflows/model-monitoring-ci.yml:35]: Looks like a lot of the commands are still windows-specific - I'll do a follow-up to convert to ubuntu once the e2e is in :)

Comment (elesel) [reply .github/workflows/model-monitoring-ci.yml:35]: Sounds good. Thanks for testing.

**Review by elesel — APPROVED** [review]

**Pull request merged**